specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/ManagedEnvironments.json

```diff
@@ -501,24 +828,167 @@
+            "Failed"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "CertificateProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "password": {
+          "description": "Certificate password.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create"
+          ]
+        },
+        "name": {
+          "description": "Friendly name of the certificate.",
+          "type": "string",
+          "readOnly": true
+        },
+        "subjectName": {
+          "description": "Subject name of the certificate.",
+          "type": "string",
+          "readOnly": true
+        },
+        "value": {
+          "format": "byte",
+          "description": "PFX or PEM blob",
+          "type": "string",
+          "x-ms-mutability": [
+            "create"
+          ]
+        },
+        "issuer": {
+          "description": "Certificate issuer.",
+          "type": "string",
+          "readOnly": true
+        },
+        "issueDate": {
+          "format": "date-time",
+          "description": "Certificate issue Date.",
+          "type": "string",
+          "readOnly": true
+        },
+        "expirationDate": {
+          "format": "date-time",
+          "description": "Certificate expiration date.",
+          "type": "string",
+          "readOnly": true
+        },
+        "thumbprint": {
+          "description": "Certificate thumbprint.",
+          "type": "string",
+          "readOnly": true
+        },
+        "valid": {
+          "description": "Is the certificate valid?.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "publicKeyHash": {
+          "description": "Public key hash.",
+          "type": "string",
+          "readOnly": true
+        },
+        "keyVaultId": {
+          "description": "Key Vault Csm resource Id.",
+          "type": "string"
+        },
+        "keyVaultSecretName": {
+          "description": "Key Vault secret name.",
+          "type": "string"
+        },
+        "keyVaultSecretStatus": {
+          "description": "Status of the Key Vault secret.",
+          "enum": [
+            "Initialized",
+            "WaitingOnCertificateOrder",
+            "Succeeded",
+            "CertificateOrderFailed",
+            "OperationNotPermittedOnKeyVault",
+            "AzureServiceUnauthorizedToAccessKeyVault",
+            "KeyVaultDoesNotExist",
+            "KeyVaultSecretDoesNotExist",
+            "UnknownError",
+            "ExternalPrivateKey",
+            "Unknown"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "KeyVaultSecretStatus",
+            "modelAsString": false
+          }
+        },
+        "canonicalName": {
+          "description": "CNAME of the certificate to be issued via free certificate",
+          "type": "string"
+        },
```

Comment:
If we are not supporting the AKV certs yet then it is better not to add corresponding properties to swagger

Reply:
I was under the impression we wanted the Swagger to be as complete as possible (even if AKV certs will not land before GA). Happy to remove it.